# Add -std=c++11 before existing CXXFLAGS, in case those already contain some -std= (which will thus
# override the preceding -std=c++11, which is likely what the caller passing in those CXXFLAGS
# intends):
#
#--- configure.ac
#+++ configure.ac
#@@ -217,7 +217,7 @@
# AS_IF([test "x$with_sharedptr" = "xc++11"], [
#     AC_MSG_CHECKING([for c++11 shared ptr])
#     save_CXXFLAGS="$CXXFLAGS"
#-    CXXFLAGS="$CXXFLAGS -std=c++11"
#+    CXXFLAGS="-std=c++11 $CXXFLAGS"
#     AC_COMPILE_IFELSE([
#         AC_LANG_PROGRAM(
#             [[#include <memory>]],
#@@ -241,7 +241,7 @@
#         AC_MSG_RESULT([no])
#     ], [
#         AC_MSG_RESULT([yes])
#-        CXXFLAGS="$CXXFLAGS -std=c++11"
#+        CXXFLAGS="-std=c++11 $CXXFLAGS"
#     ])
# ])
# AS_IF([test "x$with_sharedptr" = "xtr1"], [
--- configure
+++ configure
@@ -16723,7 +16723,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for c++11 shared ptr" >&5
 $as_echo_n "checking for c++11 shared ptr... " >&6; }
     save_CXXFLAGS="$CXXFLAGS"
-    CXXFLAGS="$CXXFLAGS -std=c++11"
+    CXXFLAGS="-std=c++11 $CXXFLAGS"
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -16780,7 +16780,7 @@
 
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-        CXXFLAGS="$CXXFLAGS -std=c++11"
+        CXXFLAGS="-std=c++11 $CXXFLAGS"
 
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
